src/overlay_12_0224E4FC.c

```diff
@@ -0,0 +1,1531 @@
+            ret = 1;
+        } else if (!loweredPriority1 && loweredPriority2) {
+            ret = 0;
+        } else if (ability1 == ABILITY_STALL && ability2 == ABILITY_STALL) {
+            if (speed1 > speed2) {
+                ret = 1;
+            } else if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else if (ability1 == ABILITY_STALL && ability2 != ABILITY_STALL) {
+            ret = 1;
+        } else if (ability1 != ABILITY_STALL && ability2 == ABILITY_STALL) {
+            ret = 0;
+        } else if (ctx->fieldCondition & FIELD_CONDITION_TRICK_ROOM) {
+            if (speed1 > speed2) {
+                ret = 1;
+            }
+            if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        } else {
+            if (speed1 < speed2) {
+                ret = 1;
+            }
+            if (speed1 == speed2 && BattleSys_Random(bsys)&1) {
+                ret = 2;
+            }
+        }
+    } else if (movePriority1 < movePriority2) {
+        ret = 1;
+    }
+    
+    return ret;
+}
+
+//Function may be mislabeled
+void BattleSystem_ClearExperienceEarnFlags(BATTLECONTEXT *ctx, int battlerId) {
+    ctx->unk_A4[(battlerId >> 1) & 1] = 0; 
+}
+
+void BattleSystem_SetExperienceEarnFlags(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+    int i = 0;
+    u32 battleType = BattleSys_GetBattleType(bsys);
+    
+    while (i <= 2) {
+        if (!(ctx->unk_3108 & MaskOfFlagNo(i)) &&
+            !(ctx->unk_3108 & MaskOfFlagNo(battlerId)) &&
+            ctx->battleMons[battlerId].hp) {
+                ctx->unk_A4[(battlerId >> 1) & 1] |= MaskOfFlagNo(ctx->selectedMonIndex[i]);
+            }
+        i += 2;
+        if (battleType == 0x4a || battleType == 0x4b) {
+            break;
+        }
+    }
+}
+
+BOOL ov12_022503EC(BattleSystem *bsys, BATTLECONTEXT *ctx, int *out) {
+    BOOL ret = FALSE;
+    
+    if (ctx->unk_2170 & (1 << 29)) {
+        *out = ov12_02258348(ctx, 1, ctx->unk_2170);
+        ctx->unk_2170 = 0;
+        if (!(ctx->moveStatusFlag & 0x801FDA49)) {
+            ret = TRUE;
+        }
+    } else if (ctx->unk_2170) {
+        *out = ov12_02258348(ctx, 1, ctx->unk_2170);
+        if (ctx->battleMons[ctx->battlerIdStatChange].hp && 
+            (!(ctx->moveStatusFlag & 0x801FDA49) || 
+            ((ctx->unk_2170 & (1 << 23)) && (ctx->moveStatusFlag & 0x40008)) ||
+            ((ctx->unk_2170 & (1 << 28)) && (ctx->moveStatusFlag & 0x10001)))) {
+                ret = TRUE;
+            }
+        ctx->unk_2170 = 0;
+    }
+    
+    return ret;
+}
+
+BOOL ov12_02250490(BattleSystem *bsys, BATTLECONTEXT *ctx, int *out) {
+    BOOL ret = FALSE;
+    u16 effectChance;
+    
+    if (ctx->unk_2174 & (1 << 29)) {
+        *out = ov12_02258348(ctx, 2, ctx->unk_2174);
+        ctx->unk_2174 = 0;
+        if (!(ctx->moveStatusFlag & 0x801FDA49)) {
```

Comment:
this seems like a completely arbitrary number. any chance for constant or explanation comment?

Reply:
tbh I have no idea what the number means, the original has it as "WAZA_STATUS_FLAG_NO_OUT," so my best guess is that it's probably a concatenation of all the move statuses that can return something? regardless, I'd rather wait until I can define it as a concatenation once I get to the move status constants